service/src/main/java/com/commercetools/pspadapter/payone/IntegrationService.java

```diff
@@ -84,16 +84,34 @@ private static void initTenantServiceResources(final TenantFactory tenantFactory
         if (StringUtils.isNotEmpty(paymentHandlerUrl)) {
             LOG.info("Register payment handler URL {}", paymentHandlerUrl);
             Spark.get(paymentHandlerUrl, (req, res) -> {
-                    final PaymentHandleResult paymentHandleResult = paymentHandler.handlePayment(req.params("id"));
-                    if (!paymentHandleResult.body().isEmpty()) {
+                    final PayoneResult payoneResult = paymentHandler.handlePayment(req.params("id"));
+                    if (!payoneResult.body().isEmpty()) {
                         LOG.debug("--> Result body of ${getTenantName()}/commercetools/handle/payments/{}: {}",
-                            req.params("id"), paymentHandleResult.body());
+                            req.params("id"), payoneResult.body());
                     }
-                    res.status(paymentHandleResult.statusCode());
+                    res.status(payoneResult.statusCode());
                     return res;
                 },
                 new HandlePaymentResponseTransformer());
         }
+        // register start Session URL
+        String startSessionUrl = tenantFactory.getPayoneStartSessionUrl();
+        if (StringUtils.isNotEmpty(startSessionUrl)) {
+            LOG.info("Register start session URL {}", startSessionUrl);
+            SessionHandler sessionHandler= tenantFactory.getSessionHandler();
+            Spark.get(startSessionUrl, (req, res) -> {
+                        final PayoneResult payoneResult = sessionHandler.startSession(req.params("id"));
```

Comment:
I checked that `req.params("id")` returns a string `id=<paymentId>`, not purely the payment ID itself. We have to add logic here to extract the `paymentId` by separate it with delimiter "="

Reply:
no, the payment-id is not attached as parameter to the url but instead it is directly attached to the url,like : 
http://localhost:9001/CAR/commercetools/start/session/<paymentid>.
So we don`t have to change the payment-id handling here.